shell/server/grain-server.js

```diff
@@ -121,10 +121,34 @@ Meteor.publish("tokenInfo", function (token) {
         let identity = globalDb.getIdentity(apiToken.owner.user.identityId);
         let metadata = apiToken.owner.user.denormalizedGrainMetadata;
         if (identity && metadata) {
+          const login = {};
+          // TODO(cleanup): This switching logic does not belong here. Perhaps it should go
+          //   in sandstorm-db/profile.js or be added as a method on Accounts.identityServices[].
+          if (identity.profile.service === "google") {
+            login.oneClick = { method: "loginWithGoogle",
```

Comment:
Hmm, specifying the exact client-side method to call in JSON from the server seems at best ugly, at worst scary.

As just one problem: Although we currently update the client and server in lockstep, eventually we may want to move towards making sure a new server is backwards-compatible with old clients, in order to avoid disrupting the user as much as we do today.

I think you should be returning a structure here that identifies the service to use by its regular service name ("google", "github", etc.) and has fields specifying any specific parameters needed to direct login, and then have a separate cascading if/else on the client side (or better yet, add a new method to the objects in `Accounts.identityServices`).